spacewalk/package/spacewalk.spec

```diff
@@ -109,20 +109,13 @@ Provides:       spacewalk-db-virtual = %{version}-%{release}
 Requires:       spacewalk-backend-sql-postgresql
 Requires:       spacewalk-java-postgresql
 Requires:       perl(DBD::Pg)
-%if 0%{?sle_version}
-%if 0%{?sle_version} >= 150400
-Requires:       postgresql14
-Requires:       postgresql14-contrib
-# we do not support postgresql versions > 14.x yet
-Conflicts:      postgresql-implementation >= 15
-Conflicts:      postgresql-contrib-implementation >= 15
-%else # not sle_version >= 150400
-Requires:       postgresql13
-Requires:       postgresql13-contrib
-# we do not support postgresql versions > 13.x yet
-Conflicts:      postgresql-implementation >= 14
-Conflicts:      postgresql-contrib-implementation >= 14
-%endif # if sle_version >= 150400
+%if 0%{?suse_version}
+# Actual version set by prjconf, default is 14
+%{!?postgresql_version: %global postgresql_version 14}
+Requires:       postgresql-implementation = %{postgresql_version}
+Requires:       postgresql-contrib-implementation = %{postgresql_version}
+Conflicts:      postgresql-implementation > %{postgresql_version}
```

Comment:
@aaannz , I have tested this and I don't think it works the way you want it to work.
You put `%{!?postgresql_version_max: %global postgresql_version_max 15}`.
This resolves to `Conflicts:      postgresql-implementation > 15`.
As soon as you have a minor version like 15.1 (or probably even 15.0.1 or 15.0-1), the conflict is triggered and you cannot install PgSQL15 anymore.

I will ponder over this one for a bit to figure out how to best do it.